Kernel/TypedTermList.hpp

```diff
@@ -20,16 +23,21 @@ class TypedTermList : public TermList
 {
   SortId _sort;
 public:
+  SortId sort() const { return _sort; }
+  TermList untyped() const { return *this; }
+  auto asTuple() const -> decltype(auto) { return std::make_tuple(untyped(), sort()); }
+  IMPL_COMPARISONS_FROM_TUPLE(TypedTermList);
+  IMPL_HASH_FROM_TUPLE(TypedTermList);
+
 
     // TODO get rid of default constructor
   TypedTermList() {}
   TypedTermList(TermList t, SortId sort) : TermList(t), _sort(sort) 
   { 
-    ASS_NEQ(sort, AtomicSort::superSort());
+    // ASS_NEQ(sort, AtomicSort::superSort());
```

Comment:
Is this no longer true?

Reply:
No because you can now call the bottom up evaluation of a `TypedTermList` with context arguments that let you either skip or not skip the type arguments of subterms. I added a test case `example_03__compute_size_with_context` in `tBottomUpEvaluation` showcasing that.